sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionClient.java

```diff
@@ -0,0 +1,219 @@
+                            connection.setRequestProperty("Proxy-Authorization", "Basic " + authStringEnc);
+                        }
+                    }
+                    if (connectionTimeout != null) {
+                        connection.setConnectTimeout((int) connectionTimeout.toMillis());
+                    }
+                }
+                return connection;
+            } catch (IOException e) {
+                throw new RuntimeException("Error opening HTTP connection", e);
+            }
+        });
+    }
+
+    // Set properties and headers on the HttpURLConnection
+    private Mono<Void> setConnectionRequest(HttpURLConnection connection, HttpRequest httpRequest) {
+        return Mono.fromRunnable(() -> {
+            try {
+                connection.setRequestMethod(httpRequest.getHttpMethod().toString());
+            } catch (ProtocolException e) {
+                throw new RuntimeException(e);
+            }
+            for (HttpHeader header : httpRequest.getHeaders()) {
+                for (String value : header.getValues()) {
+                    connection.addRequestProperty(header.getName(), value);
+                }
+            }
+            if (httpRequest.getHttpMethod() != HttpMethod.GET) {
+                connection.setDoOutput(true);
+            }
+        });
+    }
+
+    // Write the body of the request if necessary
+    private Mono<Void> writeRequestBody(HttpURLConnection connection, HttpRequest httpRequest, ProgressReporter progressReporter) {
+        switch(httpRequest.getHttpMethod()) {
+            case POST:
+            case PUT:
+            case PATCH:
+            case DELETE:
+                if (httpRequest.getBody() != null) {
+                    return Mono.fromRunnable(() -> {
+                        try (DataOutputStream os = new DataOutputStream(new BufferedOutputStream(connection.getOutputStream()))) {
+
+                            Flux<ByteBuffer> requestBody = httpRequest.getBody();
+                            if (progressReporter != null) {
+                                requestBody = requestBody.map(buffer -> {
+                                    progressReporter.reportProgress(buffer.remaining());
+                                    return buffer;
+                                });
+                            }
+
+                            requestBody
+                                .doOnNext(buffer -> {
+                                    try {
+                                        byte[] bytes = new byte[buffer.remaining()];
+                                        buffer.get(bytes);
+                                        os.write(bytes);
+                                    } catch (IOException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                }).blockLast();
```

Comment:
Should not use blocking code in async method.

Reply:
To be reviewed in next pull request.